### ISSUE TITLE
Error detected while processing function <lambda>

### ISSUE THREAD

**Issue by adisen99** [Xuyuanp/nerdtree-git-plugin#171]
#### Self-Diagnosis

- [x] I have searched the [issues](https://github.com/Xuyuanp/nerdtree-git-plugin/issues) for an answer to my question.
- [x] I have reviewed the NERDTree documentation(README.md).
- [x] I have searched the web for an answer to my question.

#### Environment (for bug reports)
- [x] Operating System: KDE Neon based on Ubuntu 20.04 (kernel 5.4.0-52-generic)
- [x] vimrc settings
    ```vim
    " Nerdtree Settings

    "let NERDTreeMinimalUI = 1
    "let NERDTreeDirArrows = 1
    map <leader>n :NERDTreeToggle<CR>
    map <leader>f :NERDTreeFind<CR>
    " Automatically start nerdtree if no file is opened
    autocmd StdinReadPre * let s:std_in=1
    autocmd VimEnter * if argc() == 0 && !exists("s:std_in") | NERDTree | endif
    " and automatically start nerdtree on openning up a directory
    autocmd VimEnter * if argc() == 1 && isdirectory(argv()[0]) && !exists("s:std_in") | exe 'NERDTree' argv()[0] | wincmd    p | ene | exe 'cd '.argv()[0] | endif
    " Close vim if nerdtree is the only instance open
    autocmd bufenter * if (winnr("$") == 1 && exists("b:NERDTree") && b:NERDTree.isTabTree()) | q | endif
    ```
    - Other NERDTree-dependent Plugins
        - [ ] jistr/vim-nerdtree-tabs
        - [x] ryanoasis/vim-devicons
        - [ ] tiagofumo/vim-nerdtree-syntax-highlight
        - [ ] Others (specify):
    - [x] I've verified the issue occurs with only `nerdtree-git-plugin` installed.
- [ ] Copy-Paste `call gitstatus#doctor#Say()` outputs

#### Steps to Reproduce the Issue
1. After editing a file in a  GitHub project and quitting the vim console using `:wq`, the following is seen on the terminal-

```sh
Error detected while processing function <lambda>15[1]..407[5]..409[2]..<SNR>59_onGitStatusSuccessCB[10]..384[17]..NERDTreeRender[1]..nerdtree#renderView[1]..332[1]..353:
line   16:
E440: undo line missing
Error detected while processing function <lambda>15[1]..407[5]..409:
line    2:
E171: Missing :endif
Error detected while processing function <lambda>15[1]..407:
line    5:
E171: Missing :endif% 
```
The file is saved properly but this error above comes up each time. The error doesn't arise when using `:w` and `:q` separately to quit the file.

2. I realize this issue is available in the FAQ on the main `README.md` file of [nerdtree-git-plugin](https://github.com/Xuyuanp/nerdtree-git-plugin) for the `fish` shell, and I am using `zsh`. Despite that, I used the solution provided (adding `set shell=sh` in the `neovim` config file) but it didn't solve the issue. **NOTE-** I am getting the issue on both `zsh` and `bash` shell.

#### Current Result (Include screenshots where appropriate.)

- The output of the command `:verbose function {407}` is -

```vim
   function 407() abort dict
        Last set from ~/.vim/plugged/nerdtree-git-plugin/autoload/gitstatus/job.vim line 48
1      let self.running = 0
2      if self.failed
3          call self.onFailed()
4      else
5          call self.onSuccess()
6      endif
   endfunction
```

- The output of the command `:verbose function {408}` is -

```vim
   function 408() abort dict
        Last set from ~/.vim/plugged/nerdtree-git-plugin/autoload/gitstatus/job.vim line 57
1      if has_key(self.opts, 'on_failed_cb')
2          call call(self.opts.on_failed_cb, [self])
3      endif
   endfunction
```

- The output of the command `:verbose function {409}` is -

```vim
   function 409() abort dict
        Last set from ~/.vim/plugged/nerdtree-git-plugin/autoload/gitstatus/job.vim line 63
1      if has_key(self.opts, 'on_success_cb')
2          call call(self.opts.on_success_cb, [self])
3      endif
   endfunction
```

#### Expected Result
The file would save without any error.

**Comment by imbible** [comment]
I have exactly the same issue, and I am using bash.

**Comment by chadostlund** [comment]
may not be the same but, I resolved an issue by removing / commenting
```vimscript
" let g:NERDTreeColorMapCustom = {
"     \ "Staged"    : "#0ee375",
"     \ "Modified"  : "#d9bf91",
"     \ "Renamed"   : "#51C9FC",
"     \ "Untracked" : "#FCE77C",
"     \ "Unmerged"  : "#FC51E6",
"     \ "Dirty"     : "#FFBD61",
"     \ "Clean"     : "#999999",
"     \ "Ignored"   : "#808080"
"     \ }
```
had no idea why. The var setting was in my vimrc. Hope this helps.

**Comment by acdifran** [comment]
@adisen99 did you find a solution to this? I'm running into the same thing.

**Comment by adisen99** [comment]
> @adisen99 did you find a solution to this? I'm running into the same thing.

Hey @acdifran, Unfortunately I didn't. There is a solution given on the FAQ on the main README page of this project, but that didn't seem to work for me. Try and that and see if it works for you. 

I actually got a new laptop and installed linux on it with zsh as the main shell and I haven't had the problem since. Hopefully you find the fix.

**Comment by Xuyuanp** [comment]
@acdifran try to upgrade your vim/nvim.

(Sorry I don't have much time to deal with this issue recently, because there are lots of work to do in my real life before the Chinese Spring Festival)

**Comment by acdifran** [comment]
@adisen99 @Xuyuanp no worries, I tried tracing the error a bit more and experimenting with a few things and it seemed to be related to trying to auto close nerdtree when it is the last window once all other buffers are closed. However, I had to restart my computer for an unrelated reason and after a reboot I'm not seeing this issue anymore. Not sure why, but it seems to have resolved itself, at least for now.
